CalendarEventCore/CalendarEvents.Services/GenericService.cs

```diff
@@ -0,0 +1,141 @@
+﻿using CalendarEvents.Models;
+using CalendarEvents.DataAccess;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using Microsoft.Extensions.Logging;
+
+namespace CalendarEvents.Services
+{
+    public interface IGenericService<T> : IGetService<T>, 
+                                      IUpdateService<T>,
+                                      IInsertService<T>,
+                                      IDeleteService
+    {
+    }
+
+    public class GenericService<T> : IGenericService<T> where T : class
+    {
+        private readonly IGenericRepository<T> _repository;
+        private readonly ILogger<GenericService<T>> _log;
+
+        public GenericService(IGenericRepository<T> repository, ILogger<GenericService<T>> log)
+        {
+            this._repository = repository;
+            this._log = log;
+        }
+
+        public ResultService Insert(T obj)
+        {
+            try
+            {
+                this._repository.Insert(obj);
+
+                return ResultService.Ok();
+            }
+            catch (Exception ex)
+            {
+                _log.LogError(ex, "Failed inserting object to DB");
```

Comment:
@cccmir Isn't it better to put the LogError from inside the `ResultService.Fail(ex);`?

Reply:
you probably right, i added the logging to quickly just to see the flow